execution/gethexec/executionengine.go

```diff
@@ -275,6 +275,59 @@ func (s *ExecutionEngine) SequenceTransactions(header *arbostypes.L1IncomingMess
 	})
 }
 
+func (s *ExecutionEngine) SequenceMessage(msg arbostypes.L1IncomingMessage, hooks *arbos.SequencingHooks) (*types.Block, error) {
+	return s.sequencerWrapper(func() (*types.Block, error) {
+		lastBlockHeader, err := s.getCurrentHeader()
+		if err != nil {
+			return nil, err
+		}
+
+		statedb, err := s.bc.StateAt(lastBlockHeader.Root)
+		if err != nil {
+			return nil, err
+		}
+
+		delayedMessagesRead := lastBlockHeader.Nonce.Uint64()
+
+		startTime := time.Now()
+		block, receipts, err := arbos.ProduceBlock(
```

Comment:
Since we are not using hooks to filter invalid transactions, won't this fail if there are any? It seems like we need to use the same hooks we were using before, and _also_ use those same hooks during replay.

This should only affect execution/computation of the state within this node, but not affect the message which gets sent to L1, because....

Reply:
It looks like the hooks are being invoked somehow by `ProduceBlockAdvanced`, and afterward the sequencer is just processing the results (where errors for individual transactions have been stored in the `hooks`). So I worry that without any hooks to catch transaction errors, the whole block will fail. But maybe you're right, the best thing to do is just try it, and it's simple enough to add the hooks back in if it becomes necessary.